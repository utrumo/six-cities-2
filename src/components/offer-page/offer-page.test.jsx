import React from 'react';
import renderer from 'react-test-renderer';
import configuraStore from 'redux-mock-store';
import {Provider} from 'react-redux';
import {BrowserRouter as Router} from 'react-router-dom';
import OfferPage from './offer-page.jsx';

const offers = [
  {
    id: 2,
    city: {
      name: `Amsterdam`,
      location: {
        latitude: 52.370216,
        longitude: 4.895168,
        zoom: 12
      }
    },
    previewImage: `img/apartment-02.jpg`,
    images: [
      `img/apartment-02.jpg`,
      `img/room.jpg`,
      `img/apartment-01.jpg`,
      `img/studio-01.jpg`,
      `img/apartment-03.jpg`,
      `img/apartment-01.jpg`
    ],
    title: `Beautiful & luxurious apartment at great location`,
    description: `Peaceful studio in the most wanted area in town. Quiet house Near of everything. Completely renovated. Lovely neighbourhood, lot of trendy shops, restaurants and bars in a walking distance.`,
    location: {
      latitude: 52.3809553943508,
      longitude: 5.939309666406198,
      zoom: 8
    },
    isPremium: true,
    rating: 4.6,
    price: 120,
    type: `apartment`,
    bedrooms: 3,
    maxAdults: 4,
    goods: [
      `Washer`,
      `Breakfast`,
      `Air conditioning`,
      `Laptop friendly workspace`,
      `Towels`,
      `Baby seat`
    ],
    host: {
      name: `Max`,
      isPro: false,
      avatarUrl: `img/avatar-max.jpg`
    }
  }
];
const offersReviews = [
  {
    id: 2,
    comments: [{
      id: 2,
      user: {
        id: 13,
        isPro: false,
        name: `Zak`,
        avatarUrl: `/img/avatar-alex.jpg`
      },
      rating: 4,
      comment: `The deluxe room was a quite comfortable one with all the adequate facilities. The only thing that made me feel uncomfortable was the rude behavior of an impolite staff at the reception desk.`,
      date: `2019-11-20T13:49:50.988Z`
    }]
  }
];
const mockStore = configuraStore();
const store = mockStore({
  currentLocation: offers[0].city.name,
  offers,
  currentOfferId: offers[0].id,
  offersReviews
});

it(`shold correct renderer after relaunch`, () => {
  const tree = renderer
    .create(
        <Provider store={store}>
          <Router>
            <OfferPage />
          </Router>
        </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
