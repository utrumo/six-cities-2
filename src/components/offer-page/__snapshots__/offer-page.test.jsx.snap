// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shold correct renderer after relaunch 1`] = `
<div
  className="page"
>
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            className="header__logo-link"
            href="/"
            onClick={[Function]}
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height="41"
              src="/img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <li
              className="header__nav-item user"
            >
              <a
                className="header__nav-link header__nav-link--profile"
                href="/login"
                onClick={[Function]}
              >
                <div
                  className="header__avatar-wrapper user__avatar-wrapper"
                />
                <span
                  className="header__login"
                >
                  Sign in
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
  <main
    className="page__main page__main--property"
  >
    <section
      className="property"
    >
      <div
        className="property__gallery-container container"
      >
        <div
          className="property__gallery"
        >
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="/img/apartment-02.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="/img/room.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="/img/apartment-01.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="/img/studio-01.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="/img/apartment-03.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="/img/apartment-01.jpg"
            />
          </div>
        </div>
      </div>
      <div
        className="property__container container"
      >
        <div
          className="property__wrapper"
        >
          <div
            className="property__mark"
          >
            <span>
              Premium
            </span>
          </div>
          <div
            className="property__name-wrapper"
          >
            <h1
              className="property__name"
            >
              Beautiful & luxurious apartment at great location
            </h1>
            <p
              className="place-card__type"
              style={
                Object {
                  "fontSize": "16px",
                  "padding": "5px 0 15px",
                  "textAlign": "center",
                }
              }
            >
              Apartment
            </p>
            <button
              className="property__bookmark-button button"
              type="button"
            >
              <svg
                className="property__bookmark-icon"
                height="33"
                width="31"
              >
                <use
                  xlinkHref="#icon-bookmark"
                />
              </svg>
              <span
                className="visually-hidden"
              >
                To bookmarks
              </span>
            </button>
          </div>
          <div
            className="property__rating rating"
          >
            <div
              className="property__stars rating__stars"
            >
              <span
                className="rating__gotten-stars"
                style={
                  Object {
                    "width": "100%",
                  }
                }
              />
              <span
                className="visually-hidden"
              >
                Rating
              </span>
            </div>
            <span
              className="property__rating-value rating__value"
            >
              5
            </span>
          </div>
          <ul
            className="property__features"
          >
            <li
              className="property__feature property__feature--entire"
            >
              Entire place
            </li>
            <li
              className="property__feature property__feature--bedrooms"
            >
              3
               Bedrooms
            </li>
            <li
              className="property__feature property__feature--adults"
            >
              Max 
              4
               adults
            </li>
          </ul>
          <div
            className="property__price"
          >
            <b
              className="property__price-value"
            >
              €
              120
            </b>
            <span
              className="property__price-text"
            >
               night
            </span>
          </div>
          <div
            className="property__inside"
          >
            <h2
              className="property__inside-title"
            >
              What's inside
            </h2>
            <ul
              className="property__inside-list"
            >
              <li
                className="property__inside-item"
              >
                Washer
              </li>
              <li
                className="property__inside-item"
              >
                Breakfast
              </li>
              <li
                className="property__inside-item"
              >
                Air conditioning
              </li>
              <li
                className="property__inside-item"
              >
                Laptop friendly workspace
              </li>
              <li
                className="property__inside-item"
              >
                Towels
              </li>
              <li
                className="property__inside-item"
              >
                Baby seat
              </li>
            </ul>
          </div>
          <div
            className="property__host"
          >
            <h2
              className="property__host-title"
            >
              Meet the host
            </h2>
            <div
              className="property__host-user user"
            >
              <div
                className="property__avatar-wrapper user__avatar-wrapper"
              >
                <img
                  alt="Host avatar"
                  className="property__avatar user__avatar"
                  height="74"
                  src="/img/avatar-max.jpg"
                  width="74"
                />
              </div>
              <span
                className="property__user-name"
              >
                Max
              </span>
            </div>
            <div
              className="property__description"
            >
              <p
                className="property__text"
              >
                Peaceful studio in the most wanted area in town. Quiet house Near of everything. Completely renovated. Lovely neighbourhood, lot of trendy shops, restaurants and bars in a walking distance.
              </p>
            </div>
          </div>
          <section
            className="property__reviews reviews"
          >
            <h2
              className="reviews__title"
            >
              Reviews ·
              <span
                className="reviews__amount"
              >
                1
              </span>
            </h2>
            <ul
              className="reviews__list"
            >
              <li
                className="reviews__item"
              >
                <div
                  className="reviews__user user"
                >
                  <div
                    className="reviews__avatar-wrapper user__avatar-wrapper"
                  >
                    <img
                      alt="Reviews avatar"
                      className="reviews__avatar user__avatar"
                      height="54"
                      src="/img/avatar-alex.jpg"
                      width="54"
                    />
                  </div>
                  <span
                    className="reviews__user-name"
                  >
                    Zak
                  </span>
                </div>
                <div
                  className="reviews__info"
                >
                  <div
                    className="reviews__rating rating"
                  >
                    <div
                      className="reviews__stars rating__stars"
                    >
                      <span
                        style={
                          Object {
                            "width": "80%",
                          }
                        }
                      />
                      <span
                        className="visually-hidden"
                      >
                        Rating
                      </span>
                    </div>
                  </div>
                  <p
                    className="reviews__text"
                  >
                    The deluxe room was a quite comfortable one with all the adequate facilities. The only thing that made me feel uncomfortable was the rude behavior of an impolite staff at the reception desk.
                  </p>
                  <time
                    className="reviews__time"
                    dateTime="2019-11-20"
                  >
                    November 2019
                  </time>
                </div>
              </li>
            </ul>
            <form
              action="#"
              className="reviews__form form"
              method="post"
            >
              <label
                className="reviews__label form__label"
                htmlFor="review"
              >
                Your review
              </label>
              <div
                className="reviews__rating-form form__rating"
              >
                <input
                  className="form__rating-input visually-hidden"
                  id="5-stars"
                  name="rating"
                  type="radio"
                  value="5"
                />
                <label
                  className="reviews__rating-label form__rating-label"
                  htmlFor="5-stars"
                  title="perfect"
                >
                  <svg
                    className="form__star-image"
                    height="33"
                    width="37"
                  >
                    <use
                      xlinkHref="#icon-star"
                    />
                  </svg>
                </label>
                <input
                  className="form__rating-input visually-hidden"
                  id="4-stars"
                  name="rating"
                  type="radio"
                  value="4"
                />
                <label
                  className="reviews__rating-label form__rating-label"
                  htmlFor="4-stars"
                  title="good"
                >
                  <svg
                    className="form__star-image"
                    height="33"
                    width="37"
                  >
                    <use
                      xlinkHref="#icon-star"
                    />
                  </svg>
                </label>
                <input
                  className="form__rating-input visually-hidden"
                  id="3-stars"
                  name="rating"
                  type="radio"
                  value="3"
                />
                <label
                  className="reviews__rating-label form__rating-label"
                  htmlFor="3-stars"
                  title="not bad"
                >
                  <svg
                    className="form__star-image"
                    height="33"
                    width="37"
                  >
                    <use
                      xlinkHref="#icon-star"
                    />
                  </svg>
                </label>
                <input
                  className="form__rating-input visually-hidden"
                  id="2-stars"
                  name="rating"
                  type="radio"
                  value="2"
                />
                <label
                  className="reviews__rating-label form__rating-label"
                  htmlFor="2-stars"
                  title="badly"
                >
                  <svg
                    className="form__star-image"
                    height="33"
                    width="37"
                  >
                    <use
                      xlinkHref="#icon-star"
                    />
                  </svg>
                </label>
                <input
                  className="form__rating-input visually-hidden"
                  id="1-star"
                  name="rating"
                  type="radio"
                  value="1"
                />
                <label
                  className="reviews__rating-label form__rating-label"
                  htmlFor="1-star"
                  title="terribly"
                >
                  <svg
                    className="form__star-image"
                    height="33"
                    width="37"
                  >
                    <use
                      xlinkHref="#icon-star"
                    />
                  </svg>
                </label>
              </div>
              <textarea
                className="reviews__textarea form__textarea"
                id="review"
                name="review"
                placeholder="Tell how was your stay, what you like and what can be improved"
              />
              <div
                className="reviews__button-wrapper"
              >
                <p
                  className="reviews__help"
                >
                  To submit review please make sure to set
                  <span
                    className="reviews__star"
                  >
                    rating
                  </span>
                  and describe your stay with at least
                  <b
                    className="reviews__text-amount"
                  >
                    50 characters
                  </b>
                  .
                </p>
                <button
                  className="reviews__submit form__submit button"
                  disabled=""
                  type="submit"
                >
                  Submit
                </button>
              </div>
            </form>
          </section>
        </div>
      </div>
      <div
        className="container"
        style={
          Object {
            "paddingLeft": "initial",
            "paddingRight": "initial",
          }
        }
      >
        <section
          className="property__map map"
        >
          <div
            id="map"
            style={
              Object {
                "height": "100%",
              }
            }
          />
        </section>
      </div>
    </section>
    <div
      className="container"
    >
      <section
        className="near-places places"
      >
        <h2
          className="near-places__title"
        >
          Other places in the neighbourhood
        </h2>
        <div
          className="places__list near-places__list"
        />
      </section>
    </div>
  </main>
</div>
`;
